Problem1.java

```diff
@@ -1 +1,46 @@
 
+
+public class week1_Interview1 {
+	
+	int binarySearch(int arr[]) 
+    { 
+        int l = 0, r = arr.length - 1; 
+        int m = l + (r - l) / 2; 
+        System.out.println(m);
+        int left = m-1, right = m+1;
+        if(arr[left]!=arr[m]-1) return arr[m]-1;
+        if(arr[right]!=arr[m]+1) return arr[m]+1;
+        while (l <= r) { 
```

Comment:
This condition is always true as you are never updating l and r.
We have already discussed the solution.Work on that.
Good Luck.